R/rules-spacing.R

```diff
@@ -144,29 +151,37 @@ set_space_between_levels <- function(pd_flat) {
 #' @param force_one Wheter or not to force one space or allow multiple spaces
 #'   after the regex "^#+'*".
 start_comments_with_space <- function(pd, force_one = FALSE) {
-  comments <- pd %>%
-    filter(token == "COMMENT")
+  comment_pos <- pd$token == "COMMENT"
+  if (all(!comment_pos)) return(pd)
 
-  if (nrow(comments) == 0) return(pd)
+  comments <- pd[comment_pos, ]
 
-  non_comments <- pd %>%
-    filter(token != "COMMENT")
+  non_comments <-pd[pd$token != "COMMENT", ]
 
   comments <- comments %>%
     extract(text, c("prefix", "space_after_prefix", "text"),
-            regex = "^(#+'*)( *)(.*)$") %>%
-    mutate(space_after_prefix = nchar(space_after_prefix),
-           space_after_prefix = set_spaces(space_after_prefix, force_one),
-           text = paste0(prefix, rep_char(" ", space_after_prefix), text),
-           short = substr(text, 1, 5)) %>%
-    select(-space_after_prefix, -prefix)
-  bind_rows(comments, non_comments) %>%
+            regex = "^(#+'*)( *)(.*)$")
+  comments$space_after_prefix <- nchar(comments$space_after_prefix)
+  comments$space_after_prefix <- set_spaces(
+    comments$space_after_prefix,
+    force_one
+  )
+
+  comments$text <- paste0(
+    comments$prefix,
+    rep_char(" ", comments$space_after_prefix), comments$text
+  )
+  comments$short <- substr(comments$text, 1, 5)
+
+  comments[, setdiff(names(comments), c("space_after_prefix", "prefix"))] %>%
+    bind_rows(non_comments) %>%
     arrange(line1, col1)
 }
 
 
 set_space_before_comments <- function(pd_flat) {
   comment_after <- pd_flat$token == "COMMENT"
+  if (all(!comment_after)) return(pd_flat)
```

Comment:
Ok, Can try that. Had it before, but then I felt it's less legible, but did not think in terms of speed.